packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts

```diff
@@ -168,10 +225,17 @@ export default class MongoMemoryReplSet extends EventEmitter {
    */
   async start(): Promise<void> {
     log('start');
-    if (this._state !== 'stopped') {
-      throw new Error(`Already in 'init' or 'running' state. Use debug for more info.`);
+    switch (this._state) {
+      // case MongoMemoryReplSetStateEnum.init:
+      //   return this.waitUntilRunning();
+      case MongoMemoryReplSetStateEnum.stopped:
+        break;
+      case MongoMemoryReplSetStateEnum.running:
+      default:
+        throw new Error('Already in "init" or "running" state. Use debug for more info.');
     }
-    this.emit((this._state = 'init'));
+    this.emit((this._state = MongoMemoryReplSetStateEnum.init)); // this needs to be executed before "setImmediate"
+    await ensureAsync();
```

Comment:
I'll read about this tomorrow.
Thanks for the link https://snyk.io/blog/nodejs-how-even-quick-async-functions-can-block-the-event-loop-starve-io/ 🙏

Reply:
just to give an small example:
imagine running `start`, but not awaiting it and instead move on to `waitUntilRunning` and await that:
```ts
const replSet = new MongoMemoryReplSet();
replSet.start();
await replSet.waitUntilRunning();
```
but before the promise of `start` (with `ensureAsync`) would get executed, the promise of `waitUntilRunning` (without `ensureAsync`) will be executed first, and so the error of being in the `stopped` state would come, before `start` even had the chance to change it
-> and `start` needs to have `ensureAsync` to execute surrounding code, if not being awaited (like for spinning up multiple replSets or setting up event listeners(after the call))
(Note: this might not fully represent what is happening, but should give an "clear" example of what this is trying to prevent)